### DISCUSSION TITLE
Support all possible paths for input parameters (basedir, workdir, upperdir)?

### DISCUSSION THREAD

**Discussion by eriksjolund** [composefs/composefs#271] [General]
Supporting all possible paths, is it a goal worth aiming for?

When I was reading the code in the file _libcomposefs/lcfs-mount.c_, I noticed that
commas are escaped but not the escape character (backslash). Often such code
would also escape the escape character (i.e a backslash), something like this

```diff
--- a/libcomposefs/lcfs-mount.c
+++ b/libcomposefs/lcfs-mount.c
@@ -157,7 +157,7 @@ static void escape_mount_option_to(const char
*str, char *dest)

        d = dest + strlen(dest);
        for (s = str; *s != 0; s++) {
-               if (*s == ',')
+               if (*s == ',' || *s == '\\')
                        *d++ = '\\';
                *d++ = *s;
        }
@@ -171,7 +171,7 @@ static char *escape_mount_option(const char *str)
        int n_escapes = 0;

        for (s = str; *s != 0; s++) {
-               if (*s == ',')
+               if (*s == ',' || *s == '\\')
                        n_escapes++;
        }
```

Anyway, it seems to be a bit tricky to support all possible paths (i.e. the whole "value space" of possible paths for basedir, upperdir and workdir).

I'm bringing up this discussion topic because changes to the user interface are easier to make early in a software project. (I don't need this feature myself)

**Comment by cgwalters** [comment]
On Tue, Apr 2, 2024, at 2:06 AM, Erik Sjölund wrote:
> Supporting all possible paths, is it a goal worth aiming for?

If we can, but without deeply analyzing this I’d be fine to start by rejecting characters like \ that we think may be problematic.  Let’s be realistic no one is going to use \ in their mount points outside of a fuzzer or something, and we should just not crash or create ill defined behavior.

**Reply by eriksjolund** [reply]
I thought it would be easier for other software projects to integrate with composefs if they could use any path (i.e any possibilities of backslash, colon and comma).
If there would be a complicated escaping scheme for the composefs input parameters, then composefs could help out with some extra command-line tool to convert a path into an escaped path.

This is just my gut feeling. I'm not speaking out of experience here.
